vmaas_sync/metrics_cyndi.go

```diff
@@ -0,0 +1,163 @@
+package vmaas_sync //nolint:golint,stylecheck
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"encoding/json"
+	"fmt"
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"reflect"
+	"time"
+)
+
+var (
+	cyndiSystemsCnt = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "patchman_engine",
+		Subsystem: "vmaas_sync",
+		Name:      "cyndi_systems",
+	}, []string{"type"})
+
+	cyndiTagsCnt = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "patchman_engine",
+		Subsystem: "vmaas_sync",
+		Name:      "cyndi_tags_count",
+	}, []string{"type"})
+)
+
+type InventoryHosts []struct {
+	DisplayName   string
+	Tags          string
+	SystemProfile string
+}
+
+//Firstly, this method unmarshal jsonb fields from DB, and loads them into relevant struct
+//This method also exposes metrics regarding max/min tags per system and number of systestems
+//with/without tags and counts systems with sap_system=true
+func updateCyndiData() {
+	count, err := getCyndiData()
+	if err != nil {
+		utils.Log("err", err.Error()).Error("unable to update cyndi metrics")
+		return
+	}
+
+	for key, value := range count {
+		cyndiTagsCnt.WithLabelValues(key).Set(float64(value))
+	}
+}
+
+//Loads display_name, tags and system_profile from inventory.hosts
+//and loads them in InventoryHosts struct.
+//Tags are parsed into from jsonb string to TagSlice struct, which contains struct Tag with individual tags.
+//All tags are then stored in uniqueTagsSlice variable to compute, which tags are unique
+func getCyndiData() (map[string]int, error) {
+	count := map[string]int{}
+	uniqueTagsSlice := []models.TagSlice{}
```

Comment:
This is in effect `[][]models.Tag`. is this the intention ? If not you can just use `models.TagSlice` and `append(uniqueTagsSlice,  tags...) down the line`